Duke-MatSci/materialsmine
# Managed Services Update

**Tasks:**
1. Endpoint Development:
   - Develop a new RESTful endpoint for Managed Services that will handle all database operations. This endpoint will replace the current disparate database calls.
2. Schema Migration:
   - Migrate existing schema definitions for all collections from the MGS database to the MGI database. Ensure that the transition is seamless and data integrity is maintained.
3. Security Enhancements:
   - Implement the same internal flag feature for security that is currently used in calls to Managed Services. This feature should now also be applied to calls from Managed Services to the new REST endpoint to ensure secure data transactions.
4. Testing and Validation:
   - Add comprehensive unit tests for the new endpoint and migrated schemas.
   - Ensure that code coverage meets our standard requirements. Review and enhance coverage if necessary.
5. Remove MongoDB Configurations:
   - Eliminate all MongoDB configurations from the Managed Services setup.
6. Code Refactoring and Cleanup:
   - Delete and refactor any code related to MongoDB database creation and operations (such as insert, delete) within Managed Services.
   - Ensure that any redundant or obsolete code is removed effectively.
7. Update Docker Compose Configuration:
   - Remove MongoDB configuration entries from the Docker Compose environment variables to prevent any unintentional connections to MongoDB.
8. Library Dependencies:
   - Uninstall and remove the PyMongo library from the project as it will no longer be needed.
9. Refactor Database Operations:
   - Redirect all existing database operations to utilize a RESTful endpoint designed to handle database operations for Managed Services. This change should ensure that all database interactions are managed centrally and more efficiently.

**Acceptance Criteria:**
- The MGS database is fully decommissioned with all operations smoothly transitioned to the MGI database.
- The new endpoint for Managed Services is fully functional and integrates seamlessly with existing systems.
- Security protocols are correctly implemented and tested.
- Unit tests confirm the stability and reliability of the new setup, and code coverage metrics are satisfactory.
- MongoDB configurations and dependencies are completely removed from Managed Services.
- All database operations are successfully routed through the new RESTful endpoint without errors.
- The Managed Services container should start without encountering the `RLIMIT_NPROC` error.
- Ensure that the refactored system is fully functional and passes all existing integration tests.
